src/cpp/session/modules/SessionSource.R

```diff
@@ -184,6 +184,8 @@
    
    if (length(args) > 0)
    {
+      # Filter empty string https://github.com/rstudio/rstudio/issues/5285
+      args <- args[sapply(args, function(x) nchar(as.character(as.expression(x))) > 0 )]
```

Comment:
`as.expression(x)` might return a vector of length > 1, so this isn't the right test.

Assuming I understand the issue correctly, I think you just want to check something like `if (!is.null(ee))` below in the for loop.

Reply:
No, parse(text=" ") is expression() and thus it is unequal to null and (!is.null(ee)) does not work to filter out the empty string.

Each args element is exactly one expression.
args = str2expression(text="mtcars[,1]\n print(cat('ER'))")
> args[[1]]
mtcars[, 1]
> args[[2]]
print(cat("ER"))

The args elements can again be lists:
> args[[2]][[1]]
print
> as.expression(args[[2]])
expression(print(cat("ER")))
> as.character(as.expression(args[[2]]))
[1] "print(cat(\"ER\"))"

As these expressions are recursive lists, the casting as.expression reduces it into a single expression. I am not aware, how a vector of length > 1 could be returned.

Reply:
> No, parse(text=" ") is expression() and thus it is unequal to null and (!is.null(ee)) does not work to filter out the empty string.

Then we should properly test for this value. (`is.null()` was an example based on my expectation that the value might be `NULL`).

It looks like the problematic value here is the missing value (identical to `quote(expr = )`) so I believe the right fix here is something more like:

```
for (i in seq_along(args))
   if (!identical(args[[i]], quote(expr = )))
      freeVars <- c(freeVars, codetools:::walkCode(args[[i]], w))
```

Relying on the coercion of `as.character(as.expression())` seems both dangerous and expensive.

Reply:
Updated accordingly :)